rs/bitcoin/ckbtc/minter/src/state/eventlog.rs

```diff
@@ -332,8 +335,12 @@ pub fn replay<I: CheckInvariants>(
                 amount,
                 ..
             } => {
-                if let Err(Overdraft(overdraft)) = state.distribute_kyt_fee(kyt_provider, amount) {
-                    return Err(ReplayLogError::InconsistentLog(format!("Attempted to distribute {amount} to {kyt_provider}, causing an overdraft of {overdraft}")));
+                if Some(kyt_provider) != previous_kyt_principal {
+                    if let Err(Overdraft(overdraft)) =
+                        state.distribute_kyt_fee(kyt_provider, amount)
```

Comment:
> was only required when replaying testnet events that had empty kyt_provider field, in which case the old code was using kyt_principal as a default substitute.

But for testnet we didn't do KYT before, meaning that there shouldn't be any `DistributedKytFee` event for testnet, no? Would it be maybe more readable if we assume that inside this event we are using Mainnet so that we don't need "this hack of remembering `previous_kyt_principal`"?

Reply:
Well, the testnet setup still requires a KYT canister that is set to AcceptAll. And kyt_fee is still deducted from deposit or withdrawal transactions. It is just that `kyt_provider` was set to None, and defaults to `kyt_principal`. The previous logic would still produce all events including `CheckedUtxo` and `DistributedKytFee`.

The purpose of the changes here (and above) is to
1. Do not supply a default value to to kyt_provider when it is None.
2. Be compatible when replaying existing mainnet & testnet events.